src/sentry/south_migrations/0501_auto__add_index_group_project_id.py

```diff
@@ -0,0 +1,1491 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+from sentry.utils.db import is_postgres
+
+class Migration(SchemaMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = True
+
+    def forwards(self, orm):
+        # Adding index on 'Group', fields ['project', 'id'] to have
+        # project_id in the replication identity
+        index_name = db.create_index_name('sentry_groupedmessage', [u'project_id', 'id'])
+        if is_postgres():
+            db.commit_transaction()
+            db.execute(
+                "CREATE UNIQUE INDEX CONCURRENTLY {} ON sentry_groupedmessage (project_id, id)".format(
+                    index_name,
+                )
+            )
+            db.start_transaction()
+            db.execute(
```

Comment:
Does anyone have preferences whether to run this into a transaction or not ?

Reply:
I prefer running nothing in transactions in DDLs just because it can be problematic from long running transactions.

Reply:
understood now it makes sense why we commit the transaction before doing any indexing work.